pkg/source/mtadapter/adapter.go

```diff
@@ -279,3 +286,24 @@ func (a *Adapter) ResolveSecret(ctx context.Context, ns string, ref *corev1.Secr
 	a.logger.Errorw("missing secret key or empty secret value", zap.String("secretname", ref.Name), zap.String("secretkey", ref.Key))
 	return "", fmt.Errorf("missing secret key or empty secret value (%s/%s)", ref.Name, ref.Key)
 }
+
+// adjustResponseSize ensures the sum of all Kafka clients memory usage does not exceed the container memory request.
+func (a *Adapter) adjustResponseSize() {
```

Comment:
Is this the best of what we can do? We should prefer to allocate memory resources to Kafka sources based on the number of vreplicas per Kafka source.

Reply:
Do you have a suggestion?

Reply:
not really. I thought about setting consumer.fetch.max based to the number of partitions handled by the pod in total but this seems not a good way. Once a partition from any Kafka source joins or leaves, it affects all the Kafka source.